File: Samu/Pr04/src/Paciente.java

import java.util.*;
import java.text.SimpleDateFormat;

public class Paciente {
    private static String BD_SERVER = "localhost";
    private static String BD_NAME = "GI1819";

    private int NumSS;
    private String DNI_NIE;
    private String Nombre;
    private String Apellidos;
    private String Sexo;
    private Date FechaNacimiento;
    private String Direccion;
    private String Poblacion;
    private String Provincia;
    private String CodigoPostal;
    private Pais miPais;
    private String Telefono;
    private String e_mail;

    public static List<Paciente> ListaPacientes() {
        List<Paciente> lista = new ArrayList<>();
        BD miBD = new BD(BD_SERVER, BD_NAME);

        for (Object[] tupla : miBD.Select("SELECT * FROM tPaciente;")) {
            Paciente u = new Paciente((int) tupla[0]);
            lista.add(u);
        }

        return lista;
    }

    public Paciente(int nSS) {
        BD miBD = new BD(BD_SERVER, BD_NAME);
        Object[] tupla = miBD.Select(String.format("Select * from tPaciente where NumSS='%d';",
                nSS)).get(0);
        this.NumSS = (Integer) tupla[0];
        this.DNI_NIE = (String) tupla[1];
        this.Nombre = (String) tupla[2];
        this.Apellidos = (String) tupla[3];
        this.Sexo = (String) tupla[4];
        this.FechaNacimiento = (Date) tupla[5];
        this.Direccion = (String) tupla[6];
        this.Poblacion = (String) tupla[7];
        this.Provincia = (String) tupla[8];
        this.CodigoPostal = (String) tupla[9];
        this.miPais = new Pais((String) tupla[10]);
        this.Telefono = (String) tupla[11];
        this.e_mail = (String) tupla[12];

    }

    public Paciente(int NumSS, String DNI_NIE, String Nombre, String Apellidos, String Sexo,
                    Date FechaNacimiento, String Direccion, String Poblacion, String Provincia,
                    String CodigoPostal, Pais miPais, String Telefono, String e_mail) {

        SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd");
        String DateToStr = format.format(FechaNacimiento);
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Insert(String.format("INSERT INTO tPaciente (NumSS, DNI_NIE, Nombre, Apellidos, Sexo, FechaNacimiento, Direccion," +
                "Poblacion, Provincia, CodigoPostal, Pais, Telefono, e_mail) values (%s, '%s', '%s', '%s', '%s', '%s', '%s', '%s'" +
                ", '%s', '%s', '%s', '%s', '%s')", NumSS, DNI_NIE, Nombre, Apellidos, Sexo, DateToStr, Direccion, Poblacion, Provincia,
                CodigoPostal, miPais.getCodigo(), Telefono, e_mail));

        this.NumSS = NumSS;
        this.DNI_NIE = DNI_NIE;
        this.Nombre = Nombre;
        this.Apellidos = Apellidos;
        this.Sexo = Sexo;
        this.FechaNacimiento = FechaNacimiento;
        this.Direccion = Direccion;
        this.Poblacion = Poblacion;
        this.Provincia = Provincia;
        this.CodigoPostal = CodigoPostal;
        this.miPais = miPais;
        this.Telefono = Telefono;
        this.e_mail = e_mail;

    }

    public List<Cita> getCitas() {
        return Cita.ListaCitas(this);
    }

    public void BorrarPaciente() {
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Delete(String.format("delete from tPaciente where NumSS = '%d';", this.NumSS));
    }

    public int getNumeroSS_Paciente() {
        return NumSS;
        // No se permite cambiar el Numero de la SS de un paciente
    }


    public String getDNI_NIE_Paciente() {
        return DNI_NIE;
    }

    public void setDNI_NIE_Paciente(String value) {
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Update(String.format("update tPaciente set DNI_NIE = '%s' where NumSS = '%d';", value, this.NumSS));
        this.DNI_NIE = value;
    }

    public String getNombre_Paciente() {
        return Nombre;
    }

    public void setNombre_Paciente(String value) {

    }

    public String getApellidos_Paciente() {
        return Apellidos;
    }

    public void setApellidos_Paciente(String value) {
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Update(String.format("update tPaciente set Nombre = '%s' where NumSS = '%d';", value, this.NumSS));
        this.Nombre = value;
    }

    public String getSexo_Paciente() {
        return Sexo;
    }

    public void setSexo_Paciente(String value) {
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Update(String.format("update tPaciente set Sexo = '%s' where NumSS = '%d';", value, this.NumSS));
        this.Sexo = value;
    }

    public Date getFechaNacimiento_Paciente() {
        return FechaNacimiento;
    }

    public void setFechaNacimiento_Paciente(Date value) {
        SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd");
        String DateToStr = format.format(value);
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Update(String.format("update tPaciente set FechaNacimiento = '%s' where NumSS = '%d';", DateToStr, this.NumSS));
        this.FechaNacimiento = value;
    }

    public String getDireccion_Paciente() {
        return Direccion;
    }

    public void setDireccion_Paciente(String value) {
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Update(String.format("update tPaciente set Direccion = '%s' where NumSS = '%d';", value, this.NumSS));
        this.Direccion = value;
    }

    public String getPoblacion_Paciente() {
        return Poblacion;
    }

    public void setPoblacion_Paciente(String value) {
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Update(String.format("update tPaciente set Poblacion = '%s' where NumSS = '%d';", value, this.NumSS));
        this.Poblacion = value;
    }

    public String getProvincia_Paciente() {
        return Provincia;
    }

    public void setProvincia_Paciente(String value) {
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Update(String.format("update tPaciente set Provincia = '%s' where NumSS = '%d';", value, this.NumSS));
        this.Provincia = value;
    }

    public String getCodigoPostal_Paciente() {
        return CodigoPostal;
    }

    public void setCodigoPostal_Paciente(String value) {
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Update(String.format("update tPaciente set CodigoPostal = '%s' where NumSS = '%d';", value, this.NumSS));
        this.CodigoPostal = value;
    }

    public Pais getPais_Paciente() {
        return miPais;
    }

    public void setPais_Paciente(Pais value) {
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Update(String.format("update tPaciente set Pais = '%s' where NumSS = '%d';", value.getCodigo(), this.NumSS));
        this.miPais = value;
    }

    public String getTelefono_Paciente() {
        return Telefono;
    }

    public void setTelefono_Paciente(String value) {
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Update(String.format("update tPaciente set Telefono = '%s' where NumSS = '%d';", value, this.NumSS));
        this.Telefono = value;
    }

    public String getE_mail_Paciente() {
        return e_mail;
    }

    public void setE_mail_Paciente(String value) {
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Update(String.format("update tPaciente set e_mail = '%s' where NumSS = '%d';", value, this.NumSS));
        this.e_mail = value;
    }

    public String toString() {
        return NumSS + "->" + DNI_NIE + "//" + Nombre + " " + Apellidos;
    }
}

File: Aleks/Practica6/Paciente.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using BDLibrary;

namespace HIS
{
    public class Paciente
    {
        private static string BD_SERVER = Properties.Settings.Default.BD_SERVER;
        private static string BD_NAME = Properties.Settings.Default.BD_NAME;

        private int NumSS;
        private string DNI_NIE;
        private string Nombre;
        private string Apellidos;
        private string Sexo;
        private DateTime FechaNacimiento;
        private string Direccion;
        private string Poblacion;
        private string Provincia;
        private string CodigoPostal;
        private Pais miPais;
        private string Telefono;
        private string e_mail;

        public static List<Paciente> ListaPacientes() {
            List<Paciente> lista = new List<Paciente>();
            SQLSERVERDB bd = new SQLSERVERDB(BD_SERVER, BD_NAME);
            foreach (object[] tuple in bd.Select("select numSS from tPaciente;")) {
                lista.Add(new Paciente((int) tuple[0]));
            }
            return lista;
        }

        public Paciente(int nSS) {
            SQLSERVERDB db = new SQLSERVERDB(BD_SERVER, BD_NAME);
            object[] tuple = db.Select("SELECT * FROM tPaciente WHERE numSS = '" + nSS + "';")[0];
            NumSS = (int) tuple[0];
            DNI_NIE = (string) tuple[1];
            Nombre = (string) tuple[2];
            Apellidos = (string) tuple[3];
            Sexo = (string) tuple[4];
            // solution adapted from https://stackoverflow.com/questions/919244/converting-a-string-to-datetime
            FechaNacimiento = DateTime.ParseExact((string) tuple[5], "yyyy-MM-dd",
                System.Globalization.CultureInfo.InvariantCulture);
            Direccion = (string) tuple[6];
            Poblacion = (string) tuple[7];
            Provincia = (string) tuple[8];
            CodigoPostal = (string) tuple[9];
            miPais = new Pais((string) tuple[10]);
            Telefono = (string) tuple[11];
            e_mail = (string) tuple[12];
        }

        public Paciente(int NumSS, string DNI_NIE, string Nombre, string Apellidos, string Sexo,
            DateTime FechaNacimiento, string Direccion, string Poblacion, string Provincia,
            string CodigoPostal, Pais miPais, string Telefono, string e_mail) {
            this.NumSS = NumSS;
            this.DNI_NIE = DNI_NIE;
            this.Nombre = Nombre;
            this.Apellidos = Apellidos;
            this.Sexo = Sexo;
            this.FechaNacimiento = FechaNacimiento;
            this.Direccion = Direccion;
            this.Poblacion = Poblacion;
            this.Provincia = Provincia;
            this.CodigoPostal = CodigoPostal;
            this.miPais = miPais;
            this.Telefono = Telefono;
            this.e_mail = e_mail;
            SQLSERVERDB db = new SQLSERVERDB(BD_SERVER, BD_NAME);
            db.Insert("insert into tPaciente values ('" + NumSS + "', '" + DNI_NIE + "', '" + Nombre + "','"
                      +Apellidos+"','"+Sexo+"','"+FechaNacimiento+"','"+Direccion+"','"+Poblacion+"','"+
                      Provincia+"','"+CodigoPostal+"','"+miPais.Codigo+"','"+Telefono+"','"+e_mail);
        }

        public void BorrarPaciente(){
            SQLSERVERDB db = new SQLSERVERDB(BD_SERVER, BD_NAME);
            
        }

        public int NumeroSS_Paciente {
            get { return NumSS;}
        }


        public string DNI_NIE_Paciente{
            get { return DNI_NIE; }
            set {
                SQLSERVERDB db = new SQLSERVERDB(BD_SERVER, BD_NAME);
                db.Update("update tPaciente set DNI_NIE = '" + value + "' where numss = '" + NumSS + "';");
                DNI_NIE = value;
            }
        }

        public string Nombre_Paciente {
            get { return Nombre; }
            set {
                SQLSERVERDB db = new SQLSERVERDB(BD_SERVER, BD_NAME);
                db.Update("update tPaciente set Nombre = '" + value + "' where numss = '" + NumSS + "';");
                Nombre = value;
            }
        }

        public string Apellidos_Paciente
        {
            get { return Apellidos; }
            set {
                SQLSERVERDB db = new SQLSERVERDB(BD_SERVER, BD_NAME);
                db.Update("update tPaciente set Apellidos = '" + value + "' where numss = '" + NumSS + "';");
                Apellidos = value;
            }
        }

        public string Sexo_Paciente
        {
            get { return Sexo; }
            set {
                SQLSERVERDB db = new SQLSERVERDB(BD_SERVER, BD_NAME);
                db.Update("update tPaciente set Sexo = '" + value + "' where numss = '" + NumSS + "';");
                Sexo = value;
            }
        }

        public DateTime FechaNacimiento_Paciente
        {
            get { return FechaNacimiento; }
            set {
                SQLSERVERDB db = new SQLSERVERDB(BD_SERVER, BD_NAME);
                db.Update("update tPaciente set FechaNacimiento = '" + value + "' where numss = '" + NumSS + "';");
                FechaNacimiento = value;
            }
        }

        public string Direccion_Paciente
        {
            get { return Direccion; }
            set {
                SQLSERVERDB db = new SQLSERVERDB(BD_SERVER, BD_NAME);
                db.Update("update tPaciente set Direccion = '" + value + "' where numss = '" + NumSS + "';");
                Direccion = value;
            }
        }

        public string Poblacion_Paciente
        {
            get { return Poblacion; }
            set {
                SQLSERVERDB db = new SQLSERVERDB(BD_SERVER, BD_NAME);
                db.Update("update tPaciente set Poblacion = '" + value + "' where numss = '" + NumSS + "';");
                Poblacion = value;
            }
        }

        public string Provincia_Paciente
        {
            get { return Provincia; }
            set {
                SQLSERVERDB db = new SQLSERVERDB(BD_SERVER, BD_NAME);
                db.Update("update tPaciente set Provincia = '" + value + "' where numss = '" + NumSS + "';");
                Provincia = value;
            }
        }

        public string CodigoPostal_Paciente
        {
            get {return CodigoPostal; }
            set
            {
                SQLSERVERDB db = new SQLSERVERDB(BD_SERVER, BD_NAME);
                db.Update("update tPaciente set CodigoPostal = '" + value + "' where numss = '" + NumSS + "';");
                CodigoPostal = value;
            }
        }

        public Pais Pais_Paciente
        {
            get { return miPais; }
            set{
                SQLSERVERDB db = new SQLSERVERDB(BD_SERVER, BD_NAME);
                db.Update("update tPaciente set Pais = '" + value + "' where numss = '" + NumSS + "';");
                miPais = value;
            }
        }

        public string Telefono_Paciente
        {
            get { return Telefono; }
            set{
                SQLSERVERDB db = new SQLSERVERDB(BD_SERVER, BD_NAME);
                db.Update("update tPaciente set Telefono = '" + value + "' where numss = '" + NumSS + "';");
                Telefono = value;
            }
        }

        public string e_mail_Paciente
        {
            get { return e_mail; }
            set{
                SQLSERVERDB db = new SQLSERVERDB(BD_SERVER, BD_NAME);
                db.Update("update tPaciente set e_mail = '" + value + "' where numss = '" + NumSS + "';");
                e_mail = value;
            }
        }

        public override string ToString()
        {
            return NumSS + "->" + DNI_NIE + "//" + Nombre + " " + Apellidos;
        }
    }
}
